server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalance.java

```diff
@@ -20,7 +20,17 @@
  *
  * @param assignments a set of the (persistent) node IDs to which each {@link ShardId} should be allocated
  */
-public record DesiredBalance(long lastConvergedIndex, Map<ShardId, ShardAssignment> assignments) {
+public record DesiredBalance(long lastConvergedIndex, Map<ShardId, ShardAssignment> assignments, ComputationFinishReason finishReason) {
+
+    enum ComputationFinishReason {
+        CONVERGED,
+        YIELD_TO_NEW_INPUT,
```

Comment:
This code is unused, right? We set the values, but we never check for them anywhere. It's kind of weird that we're not using these, and now using two parallel systems of checking `isFresh` and these result flags in `protected void processInput(DesiredBalanceInput desiredBalanceInput)` 🤨 Probably a bit of work to figure out a refactor, perhaps a TODO mentioning we should clean this up someday?